plugin/pkg/managedseed/validator/admission.go

```diff
@@ -264,6 +267,20 @@ func (v *ManagedSeed) admitGardenlet(gardenlet *seedmanagement.Gardenlet, shoot
 			allErrs = append(allErrs, errs...)
 		}
 
+		k8sVersion := semver.MustParse(shoot.Spec.Kubernetes.Version)
+		topologyAwareRoutingEnabled := gardenletConfig.FeatureGates[string(features.TopologyAwareRouting)]
+		if topologyAwareRoutingEnabled && versionutils.ConstraintK8sLess124.Check(k8sVersion) {
+			if !helper.KubeAPIServerFeatureGateEnabled(shoot, "TopologyAwareHints") {
+				allErrs = append(allErrs, field.Invalid(configPath.Child("featureGates"), gardenletConfig.FeatureGates, "the TopologyAwareRouting feature gate cannot be enabled for K8s < 1.24 clusters when the TopologyAwareHints feature gate is not enabled for kube-apiserver"))
+			}
+			if !helper.KubeControllerManagerFeatureGateEnabled(shoot, "TopologyAwareHints") {
+				allErrs = append(allErrs, field.Invalid(configPath.Child("featureGates"), gardenletConfig.FeatureGates, "the TopologyAwareRouting feature gate cannot be enabled for K8s < 1.24 clusters when the TopologyAwareHints feature gate is not enabled for kube-controller-manager"))
+			}
+			if !helper.KubeProxyFeatureGateEnabled(shoot, "TopologyAwareHints") {
+				allErrs = append(allErrs, field.Invalid(configPath.Child("featureGates"), gardenletConfig.FeatureGates, "the TopologyAwareRouting feature gate cannot be enabled for K8s < 1.24 clusters when the TopologyAwareHints feature gate is not enabled for kube-proxy"))
+			}
+		}
```

Comment:
Can this be also put in the "static" validation? Why is it part of the admission plugin?

Reply:
The ManagedSeed resource only contains a Shoot name `.spec.shoot.name`. The above validation requires the Shoot spec - it checks whether the `TopologyAwareHints` feature gate is enabled for kube-apiserver, kube-controller-manager and kube-proxy. In the static validation, I don't think I have the Shoot spec, I only have the Shoot name there. In the admission plugin, the Shoot spec is fetched and additional validation are performed with it.